[Bug] Consolidate validation for `Card.href` and `Page.path` (#348)
mckinsey/vizro: vizro-core/docs/pages/user-guides/card-button.md

- Insert the [`Card`][vizro.models.Card] into the `components` argument of the [`Page`][vizro.models.Page]
- Provide the `text` parameter with a title and some description

> - Provide the `href` parameter (relative or absolute URL). If it refers to a `Page` inside the `Dashboard`, you can insert the `Page.title` directly.

Reviewer: Here I think it does matter that we are correct. I'd leave as `Page.title` as the first thing you mention, since 99% of the time that's correct and easier to understand for a user than `Page.id`. But then add something afterwards saying that technically it's `Page.id` and link to https://vizro.readthedocs.io/en/latest/pages/user-guides/pages/#customizing-the-page-url.

> - Provide the `href` parameter (relative or absolute URL). In case it refers to a `Page` inside the `Dashboard`, you can insert the `Page.title` directly.

Reviewer 2: ```suggestion
- Provide the `href` parameter (relative or absolute URL). If it refers to a `Page` inside the `Dashboard`, you can insert the `Page.title` directly.
```
